src/storage.ts

```diff
@@ -45,11 +45,9 @@ export interface CreateBucketQuery {
   userProject: string;
 }
 
-export interface StorageOptions extends GoogleAuthOptions {
+export interface StorageOptions extends ServiceOptions {
   autoRetry?: boolean;
   maxRetries?: number;
-  promise?: typeof Promise;
```

Comment:
Which it likely doesn't work anymore, yanking this triggers the breaking change, which I want to avoid.  Can we just leave this here, and maybe flag it as deprecated?  To your point, I don't think it works anymore, and probably should have been removed in the last semver major.  

I think we should create a tracking issue for future breaking changes we want to make, so we can queue a few of these up as they come.

Reply:
👍🏻 Restored with note.